Build for 0.18.2 fails on MacOSX

```
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:196:
../node_modules/nan/nan_maybe_43_inl.h:220:17: error: 'CloneElementAt' is deprecated [-Werror,-Wdeprecated-declarations]
  return array->CloneElementAt(GetCurrentContext(), index);
                ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:3032:36: note: 'CloneElementAt' has been explicitly marked deprecated here
                MaybeLocal<Object> CloneElementAt(Local<Context> context,
                                   ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:202:
In file included from ../node_modules/nan/nan_new.h:189:
../node_modules/nan/nan_implementation_12_inl.h:40:29: error: 'New' is deprecated [-Werror,-Wdeprecated-declarations]
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                            ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:4007:56: note: 'New' has been explicitly marked deprecated here
  V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
                                                       ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
../node_modules/nan/nan.h:2017:15: error: 'SetAccessor' is deprecated [-Werror,-Wdeprecated-declarations]
  return obj->SetAccessor(
              ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:2751:22: note: 'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<Name> name,
                     ^
../src/libxmljs.cc:60:17: error: 'IsDead' is deprecated [-Werror,-Wdeprecated-declarations]
    if (v8::V8::IsDead())
                ^
/Users/andra/.node-gyp/6.5.0/include/node/v8.h:8666:10: note: 'IsDead' has been explicitly marked deprecated here
bool V8::IsDead() {
         ^
4 errors generated.
```

Maybe it can be helpful for someone.

- I'm trying to run this project [site-mapper](https://github.com/seifertd/site-mapper) which depends on `libxml@^0.18.0`, which in turn, installs 0.18.2.
- I'm working on MaxOS-Sierr
- I have downgraded `node-gyp@3.3.1  /usr/local/lib/node_modules/node-gyp` and `node@6.4.0`.

When running `npm install` I get the next error related with deprecated methods:

```
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlunicode.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlwriter.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xpath.o
In file included from ../vendor/libxml/xpath.c:476:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity' [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
2 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../../nan/nan.h:196:
../../nan/nan_maybe_43_inl.h:220:17: error: 'CloneElementAt' is deprecated [-Werror,-Wdeprecated-declarations]
  return array->CloneElementAt(GetCurrentContext(), index);
                ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:3012:36: note: 'CloneElementAt' has been explicitly marked deprecated here
                MaybeLocal<Object> CloneElementAt(Local<Context> context,
                                   ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../../nan/nan.h:202:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:40:29: error: 'New' is deprecated [-Werror,-Wdeprecated-declarations]
  return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                            ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:3976:56: note: 'New' has been explicitly marked deprecated here
  V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
                                                       ^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
../../nan/nan.h:2017:15: error: 'SetAccessor' is deprecated [-Werror,-Wdeprecated-declarations]
  return obj->SetAccessor(
              ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:2736:22: note: 'SetAccessor' has been explicitly marked deprecated here
                bool SetAccessor(Local<Name> name,
                     ^
../src/libxmljs.cc:60:17: error: 'IsDead' is deprecated [-Werror,-Wdeprecated-declarations]
    if (v8::V8::IsDead())
                ^
/Users/user/.node-gyp/6.4.0/include/node/v8.h:8462:10: note: 'IsDead' has been explicitly marked deprecated here
bool V8::IsDead() {
         ^
4 errors generated.
make: *** [Release/obj.target/xmljs/src/libxmljs.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
```

To fix it I needed to fix the libxml version to 0.18.0.

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,3 +21,4 @@ sudo: false
 before_install:
     - $CXX --version
     - case ${TRAVIS_NODE_VERSION} in 0.8*|0.10*) npm install -g npm@2 ;; esac
+    - npm install node-gyp -g

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -24,13 +24,20 @@ install:
   - npm config set msvs_version 2015
   - npm config set python C:\Python27\python.exe
   - npm install -g npm
+  - npm install -g node-gyp
   - npm install -g node-pre-gyp
+
+  # work around an issue with node-gyp v3.3.1 and node 4x
+  # https://github.com/nodejs/node-gyp/issues/921
+  - npm config set -g cafile=package.json
+  - npm config set -g strict-ssl=false
+  - npm config set registry http://registry.npmjs.org/
 
 build_script:
   - npm install --msvs_version=2015
 
 test_script:
-- cmd: node --expose-gc node_modules/nodeunit/bin/nodeunit test
+  - cmd: node --expose-gc node_modules/nodeunit/bin/nodeunit test
 
 after_test:
   - cmd: node-pre-gyp package 2>&1

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
         "package_name": "{node_abi}-{platform}-{arch}.tar.gz"
   },
   "description": "libxml bindings for v8 javascript engine",
-  "version": "0.18.2",
+  "version": "0.18.3",
   "scripts": {
     "install": "node-pre-gyp install --fallback-to-build --loglevel http",
     "test": "node --expose_gc ./node_modules/.bin/nodeunit test"
@@ -30,7 +30,7 @@
     "node": ">=0.8.0"
   },
   "dependencies": {
-    "node-pre-gyp": "0.6.28",
+    "node-pre-gyp": "~0.6.32",
     "bindings": "1.2.1",
     "nan": "~2.5.0"
   },

--- a/src/libxmljs.cc
+++ b/src/libxmljs.cc
@@ -51,16 +51,23 @@ void xmlMemFreeWrap(void* p)
     // our cleanup routines for libxml will be called (freeing memory)
     // but v8 is already offline and does not need to be informed
     // trying to adjust after shutdown will result in a fatal error
-#if (NODE_MODULE_VERSION > 0x000B)
+#if (NODE_MODULE_VERSION > 14)
+    if (v8::Isolate::GetCurrent() == 0 ||
+        v8::Isolate::GetCurrent()->IsDead())
+    {
+        return;
+    }
+#elif (NODE_MODULE_VERSION > 0x000B)
     if (v8::Isolate::GetCurrent() == 0)
     {
         return;
     }
-#endif
+#else
     if (v8::V8::IsDead())
     {
         return;
     }
+#endif
 
     const int diff = xmlMemUsed() - xml_memory_used;
     xml_memory_used += diff;

--- a/src/xml_attribute.cc
+++ b/src/xml_attribute.cc
@@ -23,7 +23,7 @@ XmlAttribute::New(xmlNode* xml_obj, const xmlChar* name, const xmlChar* value)
     }
 
     XmlAttribute* attribute = new XmlAttribute(attr);
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
     attribute->Wrap(obj);
     return scope.Escape(obj);
 }
@@ -39,7 +39,7 @@ XmlAttribute::New(xmlAttr* attr)
     }
 
     XmlAttribute* attribute = new XmlAttribute(attr);
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
     attribute->Wrap(obj);
     return scope.Escape(obj);
 }

--- a/src/xml_comment.cc
+++ b/src/xml_comment.cc
@@ -87,7 +87,7 @@ XmlComment::New(xmlNode* node)
     }
 
     XmlComment* comment = new XmlComment(node);
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
     comment->Wrap(obj);
     return scope.Escape(obj);
 }

--- a/src/xml_document.cc
+++ b/src/xml_document.cc
@@ -250,7 +250,7 @@ XmlDocument::New(xmlDoc* doc)
         return scope.Escape(static_cast<XmlDocument*>(doc->_private)->handle());
     }
 
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
 
     XmlDocument* document = Nan::ObjectWrap::Unwrap<XmlDocument>(obj);
 

--- a/src/xml_element.cc
+++ b/src/xml_element.cc
@@ -500,7 +500,7 @@ XmlElement::New(xmlNode* node)
     }
 
     XmlElement* element = new XmlElement(node);
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
     element->Wrap(obj);
     return scope.Escape(obj);
 }

--- a/src/xml_namespace.cc
+++ b/src/xml_namespace.cc
@@ -57,7 +57,7 @@ XmlNamespace::New(xmlNs* node)
     }
 
     XmlNamespace* ns = new XmlNamespace(node);
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
     ns->Wrap(obj);
     return scope.Escape(obj);
 }

--- a/src/xml_node.cc
+++ b/src/xml_node.cc
@@ -76,7 +76,7 @@ NAN_METHOD(XmlNode::Namespace) {
       // will create a new namespace attached to this node
       // since we keep the document around, the namespace, like the node, won't be
       // garbage collected
-      v8::Local<v8::Value> new_ns = define_namespace->NewInstance(argc, argv);
+      v8::Local<v8::Value> new_ns = Nan::NewInstance(define_namespace, argc, argv).ToLocalChecked();
       ns = Nan::ObjectWrap::Unwrap<XmlNamespace>(new_ns->ToObject());
   }
 

--- a/src/xml_text.cc
+++ b/src/xml_text.cc
@@ -207,7 +207,7 @@ XmlText::New(xmlNode* node)
     }
 
     XmlText* element = new XmlText(node);
-    v8::Local<v8::Object> obj = Nan::New(constructor_template)->GetFunction()->NewInstance();
+    v8::Local<v8::Object> obj = Nan::NewInstance(Nan::New(constructor_template)->GetFunction()).ToLocalChecked();
     element->Wrap(obj);
     return scope.Escape(obj);
 }

